### PR TITLE
Split application-package

### PR DESCRIPTION
**Pull request by epatpol**
Signed-off-by: Patrick-Jeffrey Pollo Guilbert <patrick.pollo.guilbert@ericsson.com>

### CODE REVIEW

File: packages/extension-manager/package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": "Theia - Extension Manager",
   "dependencies": {
     "@theia/application-package": "^0.3.7",
```
Comment (akosyakov) [inline packages/extension-manager/package.json]: it will be reexported transitively.

File: dev-packages/cli/package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "docs": "echo 'skip'"
   },
   "dependencies": {
-    "@theia/application-package": "^0.3.7"
+    "@theia/application-package": "^0.3.7",
```
Comment (akosyakov) [inline dev-packages/cli/package.json]: it will be reexported transitively.

File: dev-packages/application-manager/package.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  "scripts": {
+    "prepare": "yarn run clean && yarn run build",
+    "clean": "theiaext clean",
+    "build": "theiaext build",
+    "watch": "theiaext watch",
+    "test": "theiaext test",
+    "docs": "theiaext docs"
+  },
+  "dependencies": {
+    "@types/fs-extra": "^4.0.2",
```
Comment (akosyakov) [inline dev-packages/application-manager/package.json]: Dependencies are the most heavier part.
Please check which dependencies are used in application-package and only them keep there. Others keep here.
Avoid duplication of dependencies. All dependencies available to the application package will be available to the manager.

File: dev-packages/application-package/src/index.ts [hunk 4]
```diff
@@ -5,8 +5,7 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-export * from './rebuild';
 export * from './npm-registry';
 export * from './extension-package';
 export * from './application-package';
-export * from './application-package-manager';
+export * from './application-process';
```
Comment (akosyakov) [inline dev-packages/application-package/src/index.ts]: `application-process` is used only in the manager, should go to another package as well

File: dev-packages/application-package/package.json [hunk 5]
```diff
@@ -35,23 +35,12 @@
     "@types/write-json-file": "^2.2.1",
     "bunyan": "^1.8.10",
     "changes-stream": "^2.2.0",
-    "circular-dependency-plugin": "^2.0.0",
-    "copy-webpack-plugin": "^4.0.1",
-    "css-loader": "^0.28.1",
     "electron": "1.7.11",
```
Comment (akosyakov) [inline dev-packages/application-package/package.json]: should go the application manager

File: dev-packages/application-package/package.json [hunk 6]
```diff
@@ -35,23 +35,12 @@
...
-    "electron-rebuild": "^1.5.11",
-    "file-loader": "^0.11.1",
     "font-awesome-webpack": "0.0.5-beta.2",
```
Comment (akosyakov) [inline dev-packages/application-package/package.json]: should go the application manager

File: dev-packages/application-package/package.json [hunk 7]
```diff
@@ -35,23 +35,12 @@
     "@types/write-json-file": "^2.2.1",
     "bunyan": "^1.8.10",
```
Comment (akosyakov) [inline dev-packages/application-package/package.json]: should go the application manager

File: dev-packages/application-package/package.json [hunk 8]
```diff
@@ -35,23 +35,12 @@
...
-    "circular-dependency-plugin": "^2.0.0",
-    "copy-webpack-plugin": "^4.0.1",
-    "css-loader": "^0.28.1",
     "electron": "1.7.11",
-    "electron-rebuild": "^1.5.11",
-    "file-loader": "^0.11.1",
     "font-awesome-webpack": "0.0.5-beta.2",
     "fs-extra": "^4.0.2",
-    "ignore-loader": "^0.1.2",
     "less": "^2.7.2",
```
Comment (akosyakov) [inline dev-packages/application-package/package.json]: should go the application manager

File: packages/extension-manager/package.json [hunk 9]
```diff
@@ -3,7 +3,6 @@
   "version": "0.3.7",
   "description": "Theia - Extension Manager",
   "dependencies": {
-    "@theia/application-package": "^0.3.7",
```
Comment (akosyakov) [inline packages/extension-manager/package.json]: should be replaced by application-manager

File: dev-packages/application-package/package.json (hunk 5 above)
Comment (epatpol) [reply dev-packages/application-package/package.json]: Oops I had those changes, I just forgot to commit them!

**Comment by akosyakov** [comment]
@epatpol please take care of failing builds and ping me when you need a review again

**Comment by epatpol** [comment]
Thanks @akosyakov, whenever you have time, I just fixed the tests problem.

File: dev-packages/application-manager/src/application-package-manager.ts [hunk 10]
```diff
@@ -7,9 +7,9 @@
 
 import * as fs from 'fs-extra';
 import * as cp from 'child_process';
-import { ApplicationPackage, ApplicationPackageOptions } from "./application-package";
+import { ApplicationPackage, ApplicationPackageOptions } from "@theia/application-package";
 import { WebpackGenerator, FrontendGenerator, BackendGenerator } from "./generator";
-import { ApplicationProcess } from './application-process';
```
Comment (akosyakov) [inline dev-packages/application-manager/src/application-package-manager.ts]: don't use imports from '.'

File: dev-packages/application-package/package.json (hunk 5 above)
Comment (akosyakov) [reply dev-packages/application-package/package.json]: Please move `electron`, `less`, `font-awesome-webpack` dependencies to the manager.

**Comment by akosyakov** [comment]
@epatpol there are still unresolved comments left

**Comment by epatpol** [comment]
Sorry @akosyakov, should be good now!

File: dev-packages/application-manager/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
```
Comment (akosyakov) [inline dev-packages/application-manager/src/index.ts]: 2018 Ericsson

File: dev-packages/application-manager/src/application-process.ts [hunk 12]
```diff
@@ -8,7 +8,7 @@
 import * as path from 'path';
 import * as fs from 'fs-extra';
 import * as cp from 'child_process';
-import { ApplicationPackage } from './application-package';
+import { ApplicationPackage } from '@theia/application-package/lib';
```
Comment (akosyakov) [inline dev-packages/application-manager/src/application-process.ts]: minor: lib is superfluous

**Pull request merged**